File: Jonghyeok Kim/2022-06-29-Longest-Increasing-Subsequence.py
from typing import List

class Solution:
    def lengthOfLIS(self, nums: List[int]) -> int:
        num_len = len(nums)
        sub = [1]*num_len
        
        
        for idx, num in enumerate(nums[::-1]):
            if idx != 0:
                for idx2, num2 in enumerate(nums[num_len-idx:]):
                    if num2 > num:
                        sub[num_len-idx-1] = max(sub[num_len-idx-1], 1+sub[num_len-idx+idx2])
        return max(sub)

File: Chloe Gwon/2022-06-29-Longest-Increasing-Subsequence.js
var lengthOfLIS = function(nums) {
    var dp = new Array(nums.length).fill(1);
    var maxLength = 0;
    
    for (var i=nums.length-1; i>=0; i--){
        for (var j=i+1; j<nums.length; j++){
            if (nums[i]<nums[j] && dp[i]<dp[j]+1){
                dp[i] = dp[j]+1;
            }
        }
        maxLength = Math.max(maxLength, dp[i]);
    }
    
    return maxLength;
};

/*
example DP values:
0 1 0 3 2 3
          1
        2
      1
    3
  3
4
*/
